src/Mizar/Mizar.cpp

```diff
@@ -81,7 +85,9 @@ int main(int argc, char** argv) {
   QApplication::setOrganizationName("The Orbit Authors");
   QApplication::setApplicationName("Mizar comparison tool");
 
-  orbit_mizar_widgets::MizarMainWindow main_window(&bac);
+  orbit_mizar_widgets::MizarMainWindow main_window(
+      &bac, Baseline<QString>(MakeFileName(baseline_path)),
```

Comment:
Exactly. Yet, it helps more that it might seem from the call site. Check out the many LiftApply calls I have in Mizar. There the compiler makes sure I never pass e.g. `Baseline<Total>` and `Comparison<Addition>` into a function by mistake (unless I explicitly declare the intent to do so).